streampark-common/pom.xml

```diff
@@ -197,6 +197,33 @@
 
     <build>
         <plugins>
+            <plugin>
+                <groupId>net.alchim31.maven</groupId>
+                <artifactId>scala-maven-plugin</artifactId>
+                <version>4.3.0</version>
+                <configuration>
+                    <addScalacArgs>-target:jvm-${project.build.jdk}</addScalacArgs>
+                    <target>${project.build.jdk}</target>
+                    <source>${project.build.jdk}</source>
+                </configuration>
+                <executions>
+                    <execution>
+                        <id>install-scala</id>
+                        <phase>process-resources</phase>
+                        <goals>
+                            <goal>add-source</goal>
+                            <goal>compile</goal>
+                        </goals>
+                    </execution>
+                    <execution>
+                        <id>test-compile-scala</id>
+                        <phase>test-compile</phase>
+                        <goals>
+                            <goal>testCompile</goal>
+                        </goals>
+                    </execution>
+                </executions>
+            </plugin>
```

Comment:
scala-maven-plugin is repeatedly defined in many modules. Can it be defined in the root pom using pluginManagement and referenced before other submodules?

Reply:
> scala-maven-plugin is repeatedly defined in many modules. Can it be defined in the root pom using pluginManagement and referenced by other submodules?

It will make only java module build slowly.

Reply:
> > scala-maven-plugin is repeatedly defined in many modules. Can it be defined in the root pom using pluginManagement and referenced by other submodules?
> 
> It will make only java module build slowly.

Does the scala-maven-plugin have to be fully defined in each submodule for this to work?

Reply:
![1663258239652](https://user-images.githubusercontent.com/13617900/190454130-c66f8200-15d3-443f-a03a-69c5781c7a85.png)
This pr just tuning performance by solving these issues.